### PR TITLE
Enh/optimizer

### PR DESCRIPTION
**Pull request by jakobj**
This PR introduces pytorch optimizers to snes to update mu of the search distribution. instead of doing plain stochastic gradient descent, this opens the possibility to use any of the optimizers available in pytorch (https://pytorch.org/docs/stable/optim.html)

@mschmidt87 please have a look and let me know your thoughts on this

### CODE REVIEW

File: es/separable_natural_es.py [hunk 1]
```diff
@@ -65,26 +78,31 @@ def optimize(func, mu, sigma,
...
-    return {'mu': mu,
+        # update parameters of search distribution via natural
+        # gradient descent in natural coordinates
+
+        # set gradient and use optimizer to update mu
+        mu_torch.grad = torch.autograd.Variable(torch.DoubleTensor(-sigma * np.dot(utility, s)))
+        optimizer_mu.step()
+
+        # manually update sigma
+        sigma *= np.exp(learning_rate_sigma / 2. * np.dot(utility, s ** 2 - 1))
```
Comment (mschmidt87) [inline es/separable_natural_es.py:99]: Would it make sense here to update `log(sigma)` with the pytorch-optimizer? Or is that a bad idea due to numerical stability issues?

**Review by mschmidt87 — COMMENTED** [review]
Very nice. Would be nice if `torch.autograd` could automatically compute natural gradients.
